Jenkinsfile

```diff
@@ -283,110 +300,128 @@ def testBuilder(String platform, String nodeId, String workspace = null) {
+        stage ("Run dcos-cli tests") {
             sh '''
-               export PYTHONIOENCODING=utf-8; \
-               export DCOS_CONFIG=tests/data/dcos.toml; \
-               chmod 600 ${DCOS_CONFIG}; \
-               echo dcos_acs_token = \\\"${DCOS_ACS_TOKEN}\\\" >> ${DCOS_CONFIG}; \
-               cat ${DCOS_CONFIG}; \
-               unset DCOS_URL; \
-               unset DCOS_ACS_TOKEN; \
-               make test-binary'''
+               rm -rf ~/.dcos; \
+               cp /etc/hosts hosts.local; \
+               grep -q "^.* dcos.snakeoil.mesosphere.com$" hosts.local && \
+               sed -iold "s/^.* dcos.snakeoil.mesosphere.com$/${DCOS_URL} dcos.snakeoil.mesosphere.com/" hosts.local || \
+               echo ${DCOS_URL} dcos.snakeoil.mesosphere.com >> hosts.local; \
+               sudo cp ./hosts.local /etc/hosts'''
+    
+            dir('dcos-cli/cli') {
+                sh '''
+                   export PYTHONIOENCODING=utf-8; \
+                   export DCOS_CONFIG=tests/data/dcos.toml; \
+                   chmod 600 ${DCOS_CONFIG}; \
+                   echo dcos_acs_token = \\\"${DCOS_ACS_TOKEN}\\\" >> ${DCOS_CONFIG}; \
+                   unset DCOS_URL; \
+                   unset DCOS_ACS_TOKEN; \
+                   make test-binary'''
+            }
         }
-    }
-})
-
+    })
 
-builders['windows-tests'] = testBuilder('windows', 'windows', 'C:\\windows\\workspace')({
-    stage ("Run dcos-cli tests") {
-        bat '''
-            bash -c "rm -rf ~/.dcos"'''
-        bat '''
-            echo %DCOS_URL% dcos.snakeoil.mesosphere.com >> C:\\windows\\system32\\drivers\\etc\\hosts &
-            echo dcos_acs_token = \"%DCOS_ACS_TOKEN%\" >> dcos-cli\\cli\\tests\\data\\dcos.toml'''
+}
 
-        dir('dcos-cli/cli') {
+def windowsTestBuilder(String version, String templateUrl) {
+    return testBuilder('windows', version, templateUrl, 'windows', 'C:\\windows\\workspace')({
+        stage ("Run dcos-cli tests") {
+            bat '''
+                bash -c "rm -rf ~/.dcos"'''
             bat '''
-                bash -c " \
-                export PYTHONIOENCODING=utf-8; \
-                export DCOS_CONFIG=tests/data/dcos.toml; \
-                cat ${DCOS_CONFIG}; \
-                unset DCOS_URL; \
-                unset DCOS_ACS_TOKEN; \
-                make test-binary"'''
+                echo %DCOS_URL% dcos.snakeoil.mesosphere.com >> C:\\windows\\system32\\drivers\\etc\\hosts &
+                echo dcos_acs_token = \"%DCOS_ACS_TOKEN%\" >> dcos-cli\\cli\\tests\\data\\dcos.toml'''
+    
+            dir('dcos-cli/cli') {
+                bat '''
+                    bash -c " \
+                    export PYTHONIOENCODING=utf-8; \
+                    export DCOS_CONFIG=tests/data/dcos.toml; \
+                    unset DCOS_URL; \
+                    unset DCOS_ACS_TOKEN; \
+                    make test-binary"'''
+            }
         }
-    }
-})
+    })
+}
+
+def testBuilders = [:]
```

Comment:
Yes, def just declares a variable of any type. Specifically here we are defining a map (it infers it from the value).